modules/clients/src/test/java/org/apache/ignite/jdbc/thin/JdbcThinPartitionAwarenessReconnectionAndFailoverSelfTest.java

```diff
@@ -861,17 +867,23 @@ private void assertConnectionsCount(Map<UUID, JdbcThinTcpIo> ios, int expConnCnt
     /**
      * Invalidate connection to stopped node. Jdbc thin, won't detect that node has gone, until it tries to touch it. So
      * sending simple query to randomly chosen connection(socket), sooner or later, will touch dead one, and thus
-     * invalidate it. Please, pay attention, that it's better to send non-failoverable query, for example query with
-     * ';' somewhere in the middle.
+     * invalidate it.
      *
      * @param conn Connections.
      */
-    private void invalidateConnectionToStoppedNode(Connection conn) {
+    private void invalidateConnectionToStoppedNode(Connection conn) throws SQLException {
+        Map<UUID, JdbcThinTcpIo> ios = GridTestUtils.getFieldValue(conn, "ios");
+
+        int initialSize = ios.size();
+
         while (true) {
-            try (Statement stmt = conn.createStatement()) {
-                stmt.execute("select ';';");
```

Comment:
multi-statement query now reconnects silently and does not produces exceptions if there still live nodes in the cluster